automation/tests-container-utils.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash -ex
+
+function remove_container {
+    res=$?
+    [ "$res" -ne 0 ] && echo "*** ERROR: $res"
+    container_exec 'rm -rf $CONTAINER_WORKSPACE/*nmstate*.rpm' || true
+    ${CONTAINER_CMD} rm -f $CONTAINER_ID
+}
+
+function container_exec {
+    ${CONTAINER_CMD} exec $USE_TTY -i $CONTAINER_ID \
+        /bin/bash -c "cd $CONTAINER_WORKSPACE && $1"
+}
+
+function open_shell {
+    res=$?
+    [ "$res" -ne 0 ] && echo "*** ERROR: $res"
+    set +o errexit
+    container_exec 'echo "pytest tests/integration --pdb" >> ~/.bash_history'
+    container_exec 'exec /bin/bash'
+    run_exit
+}
+
+# Return 0 if file is changed else 1
+function is_file_changed {
+    git remote add upstream https://github.com/nmstate/nmstate.git
+    git fetch upstream
+    if [ -n "$TRAVIS_BRANCH" ]; then
+        git diff --exit-code --name-only upstream/$TRAVIS_BRANCH -- $1
+    else
+        git diff -exit-code --name-only upstream/master -- $1
+    fi
+
+    if [ $? -eq 0 ];then
+        return 1
+    else
+        return 0
+    fi
+}
+
+function rebuild_container_images {
+    set +x
+    if is_file_changed "$PROJECT_PATH/packaging"; then
+        if [ $CONTAINER_IMAGE == $CENTOS_IMAGE_DEV ];then
+            ${PROJECT_PATH}/packaging/build-container.sh centos8-nmstate-dev
+        elif [ $CONTAINER_IMAGE == $FEDORA_IMAGE_DEV ];then
+            ${PROJECT_PATH}/packaging/build-container.sh fedora-nmstate-dev
+        fi
+    fi
+}
+
+function remove_tempdir {
+    rm -rf "$NMSTATE_TEMPDIR"
+}
+
+function vdsm_tests {
+    trap remove_tempdir EXIT
+    git -C "$NMSTATE_TEMPDIR" clone --depth 1 https://gerrit.ovirt.org/vdsm
+    cd "$NMSTATE_TEMPDIR/vdsm"
+    ./tests/network/functional/run-tests.sh --nmstate-source="$PROJECT_PATH"
+    cd -
+}
+
+function collect_artifacts {
+    container_exec "
+      journalctl > "$CONT_EXPORT_DIR/journal.log" && \
+      dmesg > "$CONT_EXPORT_DIR/dmesg.log" && \
+      cat /var/log/openvswitch/ovsdb-server.log > "$CONT_EXPORT_DIR/ovsdb-server.log" && \
+      mv "$CONTAINER_WORKSPACE/pytest-run.log" "$CONT_EXPORT_DIR/pytest-run.log" || true
+    "
+}
+
+function prepare_container_environment {
```

Comment:
This is not prepare the container as the container is running now.

You may name them as:
 * `create_container`
 * `container_pre_test_setup`
 * `container_post_test_cleanup`

Reply:
I've renamed them to:
* `create_container`
* `container_pre_test_setup`
But IMO, `container_post_test_cleanup` does not fit as we are removing the container after the tests. Please see `remove_container` function.

Thanks!